validphys2/src/validphys/pineparser.py

```diff
@@ -254,7 +253,10 @@ def pineappl_reader(fkspec):
             "pineappl_reader is not prepared to read a hadronic fktable with no protons!"
         )
     Q0 = np.sqrt(pine_rep.muf2())
-    xgrid = pine_rep.x_grid()
+    x_grids = [pine.x_grid() for pine in pines]
+    xgrid = np.array([])
+    for grid in x_grids:
+        xgrid = np.union1d(xgrid, grid)
```

Comment:
```suggestion
    xgrid = np.array([])
    for pine in pines:
        xgrid = np.union1d(xgrid, pine.x_grid())
```

I don't think you need the intermediate list.